app/components/tag_list_input_component/tag_list_input_controller.js

```diff
@@ -3,6 +3,14 @@ import { Controller } from "stimulus";
 export default class extends Controller {
   static targets = ["tagList", "newTag", "template", "list"];
 
+  connect() {
+    // form hotkeys are enabled for "input" scope, we disable the form submit via enter
+    // on the tag input, so we can use the enter key to create new product tag
+    hotkeys("enter", { scope: "input", element: this.newTagTarget }, function () {
+      event.preventDefault();
+    });
+  }
+
```

Comment:
I'm not sure if this registers a global event handler or is scoped to the component element. If there's a JS event handler on the element, the handler will be cleaned up when the element is removed from the DOM. 
You should be able to check with browser devtools

If it's a global handler, we should clean this up when reloading the turbo frame. We can add a `disconnect()` and use hotkey's  [`unbind`](https://wangchujiang.com/hotkeys-js/#unbind).

Reply:
Hmm, sorry I just realised, maybe none of this is necessary. If this is to prevent the default action when adding a tag, can't you simply `event.preventDefault()` in `addTag(event)`?

Reply:
I see the comment is a bit confusing, hotkeys shortcut are enabled on all form element, with the input scope, see 
https://github.com/openfoodfoundation/openfoodnetwork/blob/40d4b0c81189824ec8ff17cd8682b1c29a1c3fac/app/webpacker/js/hotkeys.js#L4-L8

The code above is to prevent the "enter" key from submitting the form (ie the default action) , so we can use the "enter" key to create new tags.

> If this is to prevent the default action when adding a tag, can't you simply event.preventDefault() in addTag(event)?

I just tried that and it does the trick :facepalm: I am not sure why I though the only way was to disable hotkeys ??

Reply:
🤷 Hotkeys will be using JS event handlers under the hood, so we can interact with the event object directly.